src/txn/txn_rollback_to_stable.c

```diff
@@ -304,18 +282,34 @@ __rollback_row_ondisk_fixup_key(WT_SESSION_IMPL *session, WT_PAGE *page, WT_ROW
 #endif
 
     /* Allocate buffers for the data store and history store key. */
-    WT_RET(__wt_scr_alloc(session, 0, &key));
     WT_ERR(__wt_scr_alloc(session, 0, &hs_key));
     WT_ERR(__wt_scr_alloc(session, 0, &hs_value));
 
-    WT_ERR(__wt_row_leaf_key(session, page, rip, key, false));
-
-    /* Get the full update value from the data store. */
-    unpack = &_unpack;
-    __wt_row_leaf_value_cell(session, page, rip, NULL, unpack);
-    WT_ERR(__wt_page_cell_data_ref(session, page, unpack, &full_value));
-    WT_ERR(__wt_buf_set(session, &full_value, full_value.data, full_value.size));
-    newer_hs_durable_ts = unpack->tw.durable_start_ts;
+    // Unpack a row cell.
+    if (rip != NULL) {
+        WT_RET(__wt_scr_alloc(session, 0, &key));
+        WT_ERR(__wt_row_leaf_key(session, page, rip, key, false));
+
+        /* Get the full update value from the data store. */
+        unpack = &_unpack;
+        __wt_row_leaf_value_cell(session, page, rip, NULL, unpack);
+        WT_ERR(__wt_page_cell_data_ref(session, page, unpack, &full_value));
+        WT_ERR(__wt_buf_set(session, &full_value, full_value.data, full_value.size));
+        newer_hs_durable_ts = unpack->tw.durable_start_ts;
```

Comment:
These 3 lines are the same for both row and col. Move them outside of if check.

Reply:
Fixed.